src/sources/util/tcp.rs

```diff
@@ -172,8 +174,8 @@ fn handle_stream(
                     debug!("Start gracefull shutdown");
                     // Close our write part of TCP socket to signal the other side
                     // that it should stop writing and close the channel.
-                    if let Some(socket) = reader.get_ref().get_ref() {
-                        if let Err(error)=socket.shutdown(Shutdown::Write){
+                    if let Some(socket) = reader.get_ref().get_ref().get_ref().get_ref().get_ref() {
```

Comment:
A type annotation would be good here.

Reply:
This added in https://github.com/timberio/vector/pull/3095/commits/4e5c05c8487e540c37761522c7f7f4e763d1ea58#diff-36ca91cf07d40e3dc9a8f98f2c4a2762